src/Api.Rest/Contracts/IRawDataServiceRestClientBase.cs

```diff
@@ -75,6 +76,39 @@ public interface IRawDataServiceRestClientBase<T> where T : RawDataServiceFeatur
 		/// <param name="cancellationToken">A token to cancel the hronous operation.</param>
 		Task<byte[]> GetRawDataThumbnail( [NotNull] RawDataTargetEntityDto target, int rawDataKey, CancellationToken cancellationToken = default );
 
+		/// <summary>
+		/// Fetches a list of entries if specified raw data is an archive of known format.
+		/// </summary>
+		/// <param name="targetEntity">The <see cref="RawDataTargetEntityDto"/> that specifies the archive</param>
+		/// <param name="targetKey">The unique key that identifies the raw data object for the specified target.</param>
+		/// <param name="cancellationToken">A token to cancel the asynchronous operation.</param>
+		Task<RawDataArchiveIndexDto> GetRawDataArchiveEntries( RawDataTargetEntityDto targetEntity, int targetKey, CancellationToken cancellationToken = default );
+
+		/// <summary>
+		/// Fetches multiple lists of entries for queried raw data objects if they are archives of known format.
+		/// </summary>
+		/// <param name="query"><see cref="RawDataBulkQueryDto"/> with selectors containing targets of requested archives.</param>
+		/// <param name="cancellationToken">A token to cancel the asynchronous operation.</param>
+		/// <returns></returns>
+		Task<IEnumerable<RawDataArchiveIndexDto>> RawDataArchiveEntryQuery( RawDataBulkQueryDto query, CancellationToken cancellationToken = default );
+
+		/// <summary>
+		/// Fetches file as byte array of specified raw data if it is part of an archive of known format.
+		/// </summary>
+		/// <param name="targetEntity">The <see cref="RawDataTargetEntityDto"/> that specifies the archive</param>
+		/// <param name="targetKey">The unique key that identifies the raw data object for the specified target.</param>
+		/// <param name="fileName">The requested file.</param>
+		/// <param name="expectedMd5">The md5 check sum that is expected for the result object, used for caching.</param>
```

Comment:
Wenn wir das hier performant implementieren wollen, muss das der MD5-Hash der Zip-Archives sein. Dort haben wir den Hash in der Datenbank. Fuer den MD5-Hash des Zip-Eintrages muessen wir diesen entpacken. Das machen wir entweder bei jedem Zugriff erneut (= DOS-Angriff auf Server moeglich, weil CPU-intensiv) oder wir muessen uns die Zip-Eintraege in der Datenbank merken. Letzteres wollten wir nicht machen, glaube ich.

Fuer den Anwender kann es durchaus ueberraschend sein, das hier nicht der Hash der zurueckgegebenen Daten gemeint ist. Daher ist es wichtig, das hier herauszustellen und der MD5-Hash des Archives sollte bei den List-Operationen fuer die Archiveintraege auch immer mit zurueckgegeben werden.

Vorschlag: Variablename "expectedArchiveMd5"
The md5 check sum that is expected for the targeted archive. *NOT* the returned entry. Used for caching.

Reply:
Hab ich angepasst und auch in der REST API geändert. Bei den ListEntries Operationen ist der Hash dann immer Teil von ArchiveInfo (also RawDataInformation).